### PR TITLE
DRILL-7182: Incorrect access specifier for Join DrillRelMdDistinctRowcount

### PR DESCRIPTION
**Pull request by gparai**
@amansinha100 can you please review this PR? Thanks!

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ private Double getDistinctRowCount(DrillScanRelBase scan, RelMetadataQuery mq, D
     }
   }
 
-  public Double getDistinctRowCount(DrillJoinRelBase joinRel, RelMetadataQuery mq, ImmutableBitSet groupKey,
+  private Double getDistinctRowCount(DrillJoinRelBase joinRel, RelMetadataQuery mq, ImmutableBitSet groupKey,
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java]: Wouldn't it be better to rename this (and the other private methods here) with an 'internal' suffix to avoid potential issues in the future ?  Only the Override methods remain unchanged.   What do you think ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java [hunk 2]
```diff
@@ -92,7 +92,7 @@ public Double getDistinctRowCount(RelNode rel, RelMetadataQuery mq, ImmutableBit
         if (rel instanceof DrillScanRel) {
           // The existing Drill behavior is to only use this estimation for DrillScanRel and not ScanPrel.
           // TODO: We may potentially do it for ScanPrel (outside the scope of statistics)
-          return rel.estimateRowCount(mq) * 0.1;
+          return ((DrillScanRel)rel).estimateRowCount(mq) * 0.1;
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java]: This cast is not needed, please revert this change.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java (hunk 1 above)
Comment (gparai) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java]: Done

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java (hunk 2 above)
Comment (gparai) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdDistinctRowCount.java]: Done.

**Comment by gparai** [comment]
@amansinha100 / @vvysotskyi I have addressed your review comments. Please take a look.

**Comment by vvysotskyi** [comment]
+1
